File: core/core-domain/src/main/java/com/nmh/commerce/coupon/FixedPriceDiscountPolicy.java
package com.nmh.commerce.coupon;

import com.nmh.commerce.domain.Money;
import lombok.Getter;

@Getter
public class FixedPriceDiscountPolicy implements DiscountPolicy {
    private final Money discountPrice;

    public FixedPriceDiscountPolicy(Money discountPrice) {
        this.discountPrice = discountPrice;
    }

    @Override
    public Money calculateDiscount(Money originalPrice) {
        return originalPrice.minus(discountPrice);
    }
}

File: core/core-domain/src/main/kotlin/com/nmh/commerce/coupon/FixedPriceDiscountPolicy.kt
package com.nmh.commerce.coupon

import com.nmh.commerce.domain.Money

data class FixedPriceDiscountPolicy(
    val discountPrice: Money,
) : DiscountPolicy {
    override fun calculateDiscount(originalPrice: Money): Money = originalPrice.minus(discountPrice)
}
